### ISSUE TITLE
Reporting: Variables in reports sentences

### ISSUE THREAD

**Issue by san-ouadghiri** [serenity-js/serenity-js#210]
Hi again @jan-molak 

I'm using Task.where () and I'm trying to get a nice description. Here is the line:

```
export const AskAServiceHealthCheck = (service : any) => Task.where(`#actor wants to do a GET on the **<service>** service`,
    Get.resource(service.healthCheck),
);
```

If I change <service> by **` + service.toString() + `** I get
=> **Pharell** wants to do a GET on the **function Media() {}** service

Same result with `${service}` (as I saw that used in a reply you gave on a drag'n'drop question).

I'd like to have something like
=> **Tony** wants to do a GET on the **User** service

I'm not getting how #actor works, so it's though for me to get how I can make this happen for the service name i'm checking. Any doc, example or advise on this would help.

**Comment by jan-molak** [comment]
Hi @san-ouadghiri, great to hear from you again! :-)

Assuming `Media` is a class, you can get the name of its constructor using the  [`function.name`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/name) api:

```typescript
class Media { 
}

const m = new Media();

console.log(m.constructor.name)
=> 'Media'
```

Given that your example above already uses the [ES6 template literals](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Template_literals), you can extend it as follows:

```typescript
export const AskAServiceHealthCheck = (service : any) => 
  Task.where(`#actor checks the health of the ${ service.constructor.name } service`,
    Get.resource(service.healthCheck),
  );
```

Now, regarding the `#actor` template you've asked about. Have a look at the [activity description](https://github.com/jan-molak/serenity-js/blob/23110f7cfce612151b1b65d399fc407d211ba5b1/packages/core/src/recording/activity_description.ts#L14) and the [associated tests](https://github.com/jan-molak/serenity-js/blob/23110f7cfce612151b1b65d399fc407d211ba5b1/packages/core/spec/recording/activity_description.spec.ts#L28) as they should shed some light on this.

Basically, `#actor` has a "special" meaning. Serenity/JS will replace that token with the actor's name when the activity in question is reported.

Hope that helps!
Jan

**Comment by jan-molak** [comment]
In Serenity/JS 2.0 we'd use the following syntax:

```typescript
import { Task } from '@serenity-js/core';
import { Send, GetRequest, LastResponse } from '@serenity-js/rest';
import { Ensure, equals } from '@serenity-js/assertions';

export const EnsureServiceIsUp = (service : any) => 
  Task.where(`#actor ensures that the ${ service.constructor.name } service is up`,
    Send.a(GetRequest.to(service.healthCheckUrl)),
    Ensure.that(LastResponse.status(), equals(200)),
  );
```

See:
- https://serenity-js.org/modules/rest/
- https://serenity-js.org/modules/assertions/
- https://serenity-js.org/modules/core/
